hr_period/models/hr_payslip.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding:utf-8 -*-
+# Copyright 2015 Savoir-faire Linux. All Rights Reserved.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    hr_period_id = fields.Many2one(
+        'hr.period',
+        string='Period',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    date_payment = fields.Date(
+        'Date of Payment',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+
+    @api.multi
+    @api.constrains('hr_period_id', 'company_id')
+    def _check_period_company(self):
+        for slip in self:
+            if slip.hr_period_id:
+                if slip.hr_period_id.company_id != slip.company_id:
+                    raise UserError("""The company on the selected period must
+                                    be the same as the company on the
+                                    payslip.""")
+
+    @api.onchange('company_id', 'contract_id')
+    def onchange_company_id(self):
+        if self.company_id and self.contract_id:
+            contract = self.contract_id
+            period_obj = self.env['hr.period']
+            period = period_obj.get_next_period(self.company_id.id,
+                                                contract.schedule_pay)
+            self.hr_period_id = period.id if period else False
+
+    @api.multi
+    @api.onchange('contract_id')
+    def onchange_contract(self):
+        res = super(HrPayslip, self).onchange_contract()
+        if self.contract_id.employee_id and self.contract_id:
+            employee = self.contract_id.employee_id
+            contract = self.contract_id
+            period = self.env['hr.period'].get_next_period(
+                employee.company_id.id, contract.schedule_pay)
+            if period:
+                self.hr_period_id = period.id if period else False
+                self.name = 'Salary Slip of '+employee.name+' for '+period.name
```

Comment:
missing spaces around '+'

and the strings should be translated, so likely you should use %s rather than concatenations